src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -667,6 +697,9 @@ private static boolean isLoopAst(int ast) {
         /** Contains definitions of uninitialized variables. */
         private final Deque<DetailAST> uninitializedVariables = new ArrayDeque<>();
 
+        /** Whether there is a {@code break} or {@code return} in the scope. */
+        private boolean containsBreakOrReturn;
```

Comment:
This seems weird we treat `break` and `return` with the same variable as there is a difference between them and how they are handled.
`break` will exit switch/loop/etc, but continue on logic outside that scope.
`return` will exit the method completely (besides any `finally` blocks) and ignore any logic outside scope.
I am also going to assume that we don't have any handling of `continue` either since we didn't have `break`.

@Luolc Can you create a some more examples with just `break` and `return` by itself, and not intermingled like issue's example. Don't limit `break` to just `switch` statement. Please show some while/for/etc.
I will review regression shortly.